File: examples/Python.ExampleApplication/Hooks/Filters/shared_custom_view_filter.py
from tableau_migration import (
    ICustomView,
    ContentFilterBase,
    ContentFilterContextItem,
    FilterStatus)


class SharedCustomViewFilter(ContentFilterBase[ICustomView]):
    def filter(self, item: ContentFilterContextItem[ICustomView]) -> None:
        if item.source_item.shared == True:
            item.status = FilterStatus.SKIP

File: examples/Csharp.ExampleApplication/Hooks/Filters/SharedCustomViewFilter.cs
using Microsoft.Extensions.Logging;
using Tableau.Migration.Content;
using Tableau.Migration.Engine.Hooks.Filters;
using Tableau.Migration.Resources;

namespace Csharp.ExampleApplication.Hooks.Filters
{
    #region class
    public class SharedCustomViewFilter : ContentFilterBase<ICustomView>
    {
        public SharedCustomViewFilter(
            ISharedResourcesLocalizer localizer,
            ILogger<IContentFilter<ICustomView>> logger)
                : base(localizer, logger) { }

        public override void Filter(ContentFilterContextItem<ICustomView> item)
        {
            if (item.SourceItem.Shared)
            {
                item.Status = FilterStatus.Skip;
            }
        }
    }
    #endregion
}
